project-koku/koku
# COST-644 - Add OCPNamespaceLabelProcessor as NotImplementedError

- add namespace ocp report processor
- raises not implemented error

Testing:
A. Koku-listener (i.e. using ingress):
1. With ingress running locally, chuck this gzip at it: [cost-mgmt-536cc5dc-b91f-4251-88b6-fab1789ebfdd.tar.gz](https://github.com/project-koku/koku/files/5461227/cost-mgmt-536cc5dc-b91f-4251-88b6-fab1789ebfdd.tar.gz)

```
curl -vvvv -H "x-rh-identity: eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTIzNDUiLCAiaW50ZXJuYWwiOiB7Im9yZ19pZCI6ICI1NDMyMSJ9fX0=" -F "upload=@cost-mgmt-536cc5dc-b91f-4251-88b6-fab1789ebfdd.tar.gz;type=application/vnd.redhat.hccm.tar+tgz" -H "x-rh-insights-request-id: 52df9f748eabcfea" localhost:8080/api/ingress/v1/upload
```
2. Watch Listener logs. See these in logs:
```
koku_listener     | [2020-10-29 20:45:59,613] INFO None NotImplementedError: OCPNamespaceLabelReport not implemented yet.
koku_listener     | [2020-10-29 20:45:59,614] INFO None Processing: /var/tmp/masu/insights_local/41eedaa2-1d80-435c-ab37-91398c9c62e9/20201001-20201101/536cc5dc-b91f-4251-88b6-fab1789ebfdd_openshift_usage_report.0.csv complete.
```
The other files should also ingest normally.

B. Koku-worker
1. Create `testing/pvc_dir/insights_local/my-ocp-cluster-1/20201001-20201101`
2. unzip the tarball into this dir.
3. Create an OCP source with `cluster_id: my-ocp-cluster-1`
4. Ingest the files.
5. Watch koku-worker logs:
```
koku_worker       | [2020-10-29 21:01:04,951: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(179de8f0-e882-4854-a8c4-d5be458758aa via 5197bf38-2f18-45f6-8bc6-51c3461e8c01)] Processing Report:
koku_worker       |  schema_name: acct10001
koku_worker       |  provider: OCP
koku_worker       |  provider_uuid: 2c1a9c6b-18b2-4e42-afb9-73dca9c51900
koku_worker       |  file: /testing/pvc_dir/processing/acct10001/ocp/my-ocp-cluster-1/20201001-20201101_536cc5dc-b91f-4251-88b6-fab1789ebfdd_openshift_usage_report.0.csv
koku_worker       |  compression: PLAIN
koku_worker       |  start_date: 2020-10-01 00:00:00
koku_worker       | [2020-10-29 21:01:04,952: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(179de8f0-e882-4854-a8c4-d5be458758aa via 5197bf38-2f18-45f6-8bc6-51c3461e8c01)] Avaiable memory: 171368448 bytes (38.7%)
koku_worker       | [2020-10-29 21:01:04,972: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(179de8f0-e882-4854-a8c4-d5be458758aa via 5197bf38-2f18-45f6-8bc6-51c3461e8c01)] OCPNamespaceLabelReport not implemented yet.
koku_worker       | [2020-10-29 21:01:06,040] INFO 5197bf38-2f18-45f6-8bc6-51c3461e8c01 Removed task key 2c1a9c6b-18b2-4e42-afb9-73dca9c51900:/testing/pvc_dir/insights_local/my-ocp-cluster-1/20201001-20201101/536cc5dc-b91f-4251-88b6-fab1789ebfdd_openshift_usage_report.0.csv from cache.
```
